google-cloud-logging/src/test/java/com/google/cloud/logging/it/ITTailLogsTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.logging.it;
+
+import static com.google.cloud.logging.testing.RemoteLoggingHelper.formatForTest;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.logging.*;
+import com.google.cloud.logging.Logging.TailOption;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import org.junit.AfterClass;
+import org.junit.Test;
+
+public class ITTailLogsTest extends BaseSystemTest {
+
+  private static final String LOG_ID = formatForTest("test-tail-log-entries-log");
+  private static final MonitoredResource GLOBAL_RESOURCE =
+      MonitoredResource.newBuilder("global").build();
+
+  // @BeforeClass
+  // public static void configureWriteLogs() {
+  // }
+
+  @AfterClass
+  public static void cleanUpLogs() throws InterruptedException {
+    assertTrue(cleanupLog(LOG_ID));
+  }
+
+  @Test(timeout = 120_000) // Note: the test should not take longer than 2 min
+  public void testTailLogEntries() throws InterruptedException {
+    LogEntry testLogEntry =
+        LogEntry.newBuilder(Payload.StringPayload.of("stringPayload1"))
+            .addLabel("key1", "value1")
+            .setLogName(LOG_ID)
+            .setHttpRequest(HttpRequest.newBuilder().setStatus(200).build())
+            .setResource(GLOBAL_RESOURCE)
+            .build();
+
+    String filter = "logName=projects/" + logging.getOptions().getProjectId() + "/logs/" + LOG_ID;
+    LogEntryServerStream stream = logging.tailLogEntries(TailOption.filter(filter));
+
+    Runnable task =
+        () -> {
+          // it may take awhile for tailing session to startup on the backend
```

Comment:
is it documented somewhere on cloud.google.com (how long it might take)? If not, worth opening a bug on docs team to add this - I find it non-trivial assumption that customers of this API might get confused with

Reply:
Not exactly. The public doc states that it starts streaming new log entries but it does not define how quick it starts doing it. The delay might reach few seconds. The 10 second is placed to avoid flakiness in the test.
Mind that this API is still in Beta. They are working to improve the stream establishing time.